apps/cow-tools/src/app/milkman/(components)/TokenSelect.tsx

```diff
@@ -0,0 +1,210 @@
+          balance: BigInt(asset.balance),
+        };
+      });
+      setTokens(tokens);
+    }
+  }, [loaded]);
+
+  function filterTokenInput({
+    tokenSearchQuery,
+    token,
+  }: {
+    tokenSearchQuery: string;
+    token?: TokenInfo;
+  }) {
+    {
+      if (!token) return false;
+      const regex = new RegExp(tokenSearchQuery, "i");
+      return regex.test(Object.values(token).join(","));
+    }
+  }
+  return (
+    <div className="max-h-[30rem] divide-y divide-slate7 overflow-y-scroll text-white scrollbar-thin scrollbar-track-blue3 scrollbar-thumb-slate12">
+      <div className="flex h-full w-full flex-col items-center justify-center gap-y-4 py-4">
+        <div className="text-xl">Token Search</div>
+        <div className="flex w-full items-center px-10">
+          <input
+            type="text"
+            placeholder="Search name or paste address"
+            className="h-9 w-full appearance-none items-center justify-center rounded-l-[4px] bg-blue4 px-[10px] text-sm leading-none text-slate12 outline-none"
+            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+              setTokenSearchQuery(e.target.value)
+            }
+            value={tokenSearchQuery}
+          />
+          <div className="flex items-center h-9 rounded-r-[4px] bg-slate12 px-2 leading-none outline-none transition hover:bg-slate11 disabled:cursor-not-allowed">
+            <MagnifyingGlassIcon
+              color="rgb(31 41 55)"
+              className="font-semibold"
+              height={20}
+              width={20}
+            />
+          </div>
+        </div>
+      </div>
+      <Table color="blue">
+        <Table.HeaderRow>
+          <Table.HeaderCell>
+            <span className="sr-only">Token Logo</span>
+          </Table.HeaderCell>
+          <Table.HeaderCell>Token</Table.HeaderCell>
+          <Table.HeaderCell>Wallet Balance</Table.HeaderCell>
+        </Table.HeaderRow>
+        <Table.Body>
+          {tokens
+            .filter((token) => filterTokenInput({ tokenSearchQuery, token }))
+            .sort((a, b) =>
+              formatUnits(a!.balance, a!.decimals ? a!.decimals : 0) <
+              formatUnits(b!.balance, b!.decimals ? b!.decimals : 0)
+                ? 1
+                : -1,
+            )
+            .map((token) => {
+              if (token) {
+                return (
+                  <TokenRow
+                    key={token.address}
+                    token={token}
+                    onSelectToken={onSelectToken}
+                  />
+                );
+              }
+            })}
+        </Table.Body>
+      </Table>
+    </div>
+  );
+}
+
+function TokenRow({
+  token,
+  onSelectToken,
+}: {
+  token: TokenWalletBalance;
+  onSelectToken: (token: TokenWalletBalance) => void;
+}) {
+  return (
+    <Table.BodyRow
+      key={token.address}
+      classNames="hover:bg-blue4 hover:cursor-pointer"
+      onClick={() => onSelectToken(token)}
```

Comment:
better suggestion to not prop drill `onSelectToken` this whole component?

Reply:
The only other way that I know is by creating a context for this component

Reply:
hmmm just for this it's not worth it